modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/KeyValueGetPlan.java

```diff
@@ -142,60 +148,65 @@ public <RowT> AsyncCursor<InternalSqlRow> execute(
         List<RexNode> projectionExpr = lookupNode.projects();
         List<RexNode> keyExpressions = lookupNode.keyExpressions();
 
-        RelDataType rowType = sqlTable.getRowType(Commons.typeFactory(), requiredColumns);
-
-        Supplier<RowT> keySupplier = ctx.expressionFactory()
-                .rowSource(keyExpressions);
-        Predicate<RowT> filter = filterExpr == null ? null : ctx.expressionFactory()
-                .predicate(filterExpr, rowType);
-        Function<RowT, RowT> projection = projectionExpr == null ? null : ctx.expressionFactory()
-                .project(projectionExpr, rowType);
-
-        RowHandler<RowT> rowHandler = ctx.rowHandler();
-        RowSchema rowSchema = TypeUtils.rowSchemaFromRelTypes(RelOptUtil.getFieldTypeList(rowType));
-        RowFactory<RowT> rowFactory = rowHandler.factory(rowSchema);
-
-        RelDataType resultType = lookupNode.getRowType();
-        BiFunction<Integer, Object, Object> internalTypeConverter = TypeUtils.resultTypeConverter(ctx, resultType);
-
-        ScannableTable scannableTable = execTable.scannableTable();
-        Function<RowT, Iterator<InternalSqlRow>> postProcess = row -> {
-            if (row == null) {
-                return Collections.emptyIterator();
-            }
-
-            if (filter != null && !filter.test(row)) {
-                return Collections.emptyIterator();
-            }
+        CompletableFuture<Iterator<InternalSqlRow>> result;
 
-            if (projection != null) {
-                row = projection.apply(row);
+        try {
```

Comment:
does it make sense to move this `try-catch` on `ExecutionServiceImpl` level? This way we cover all the problem in a single place

Reply:
Thanks, try/catch moved to `ExecutionServiceImpl`